azkaban-exec-server/src/main/java/azkaban/execapp/AzkabanExecutorServer.java

```diff
@@ -273,7 +273,13 @@ private void startReportingExecMetrics() {
 
   private void initActive() throws ExecutorManagerException {
     final Executor executor;
-    final int port = this.props.getInt(ConfigurationKeys.EXECUTOR_PORT, -1);
+    final int port;
+    final boolean useSsl = props.getBoolean(ConfigurationKeys.JETTY_USE_SSL, true);
+    if (useSsl) {
+      port = this.props.getInt(ConfigurationKeys.EXECUTOR_SSL_PORT, -1);
```

Comment:
In above class, default port value is 0. Here it is -1. Can you please make it consistent?

Reply:
So the default port value of 0 is from the legacy code, where port 0 implies random selection of a port, which is what is used in the present executor-server. 
As -1 can never be a port, here it is to imply that the port is randomly selected and hence it can't be activated from within the code. Hence executor is activated using the ajax APIs. This is what we currently do in the executor-server deployment.